people-and-planet-ai/image-classification/e2e_test.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import io
+import os
+import subprocess
+import uuid
+
+from google.cloud import aiplatform
+from google.cloud import bigquery
+from google.cloud import storage
+import pytest
+
+import deploy_model
+import predict
+
+SUFFIX = uuid.uuid4().hex[0:6]
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+BUCKET_NAME = f"wildlife-insights-{SUFFIX}"
+BIGQUERY_DATASET = f"wildlife_insights_{SUFFIX}"
+BIGQUERY_TABLE = "images_database"
+MODEL_ENDPOINT = f"wildlife_insights_{SUFFIX}"
+REGION = "us-central1"
+MIN_IMAGES_PER_CLASS = 1
+MAX_IMAGES_PER_CLASS = 1
+
+# Use a pre-trained pre-existing model, training one takes too long.
+MODEL_PATH = f"projects/{PROJECT}/locations/{REGION}/models/1590773423066316800"
+
+
+@pytest.fixture(scope="session")
+def bucket_name() -> str:
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(BUCKET_NAME)
+
+    print(f"bucket_name: {repr(BUCKET_NAME)}")
+    yield BUCKET_NAME
+
+    # This test creates 1 image file per class, creating around 650 files.
+    # `bucket.delete(force=True)` does not work if the bucket has >256 files.
+    # Sequentially deleting many files with the client libraries can take
+    # several minutes, so we're using `gsutil -m` instead.
+    # subprocess.call(["gsutil", "-m", "rm", "-rf", f"gs://{BUCKET_NAME}/*"])
```

Comment:
Is the intent to keep this here?

Reply:
Outdated comment! Great catch, that was before adding the `bigquery_table` fixture, which creates a "hardcoded" small table. That was another optimization to reduce the testing times. I'll go ahead and just remove this whole comment :)